src/uwtools/scheduler.py

```diff
@@ -178,20 +190,35 @@ def get_scheduler(props: Mapping) -> JobScheduler:
             ) from error
         return scheduler(props)
 
+    def run_job(self, script_path: OptionalPath): -> bool
+        """
+        Submits a job to the scheduler.
+        """
+
+        subprocess.run(
+            f"{self.submit_command} {outpath}",
+            stderr=subprocess.STDOUT,
+            check=False,
+            shell=True,
+            )
```

Comment:
During next backlog grooming, I'll plead for [UW-369](https://jira.epic.oarcloud.noaa.gov/browse/UW-369) to get scheduled soon, before we duplicate more `subprocess.run()` calls. The version currently being copied is not great: The output of the command run in the subshell will just get dumped on `stdout` without being part of the logging output, and if the command fails, Python will march merrily on as if nothing bad happened. We can write a robust, reusable function that captures the output and (maybe, optionally) logs it, and checks the exit status of the command and does something (also potentially configurable -- maybe raise an exception, maybe maybe just `logging.error()` the output) if it fails.

Also, is the typechecker ok with the `bool` return type? This method returns `None` since there's no explicit `return` statement.

Reply:
Ah, that's left over from my limited subprocess run experience, I'm sure there's a better way!
I've gone back and forth design-wise on if we really want the scheduler to also be running, but I'm falling on the side of agreeing with this change!